tests/lib.rs

```diff
@@ -133,16 +135,31 @@ impl VirtualFileSystem {
 
             println!("\n-- {} –-\n", file.display());
 
-            if let Some(extension) = file.extension() {
-                if extension == "tgz" {
-                    continue; // TODO(amello): make packaging deterministic so we can validate the files
+            match fs::read_to_string(&expected) {
```

Comment:
can we do the comparison by file ext instead of using the errorkind?

so having two branches depending on the extension (text {*}, bin {tgz}) instead of trying one and then the other – that would also be more explicit on what / why stuff happens

eg having a `enum FileType` with a `from_extension` or similar (not worring to much about the naming but rather encoding what happens into the typesystem

Reply:
I wanted to avoid having to know about all file types ahead of time, in order to make this utility more flexible. That said, on a second thought, one issue with checking UTF8 encoding is the possibility of running into a binary file that happens to be UTF8 compatible by sheer chance.

Since we control the test files I think it's ok to go the enumeration route. Will make the change.